src/common/admin_socket_client.cc

```diff
@@ -23,19 +23,18 @@ using std::ostringstream;
 
 const char* get_rand_socket_path()
 {
-  static char *g_socket_path = NULL;
-
-  if (g_socket_path == NULL) {
```

Comment:
I think this is still needed... g_socket_path is static, so it'll be NULL only the first time this function executes during the process lifetime.